src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -49,56 +51,11 @@ public class Regex
 
         protected internal RegexRunnerFactory factory;
 
-        internal Dictionary<int, int> _caps;            // if captures are sparse, this is the hashtable capnum->index
-        internal Dictionary<string, int> _capnames;     // if named captures are used, this maps names->index
+        protected internal Hashtable caps;            // if captures are sparse, this is the hashtable capnum->index
+        protected internal Hashtable capnames;     // if named captures are used, this maps names->index
 
-        protected internal String[] capslist;              // if captures are sparse or named captures are used, this is the sorted list of names
+        protected internal string[] capslist;              // if captures are sparse or named captures are used, this is the sorted list of names
         protected internal int capsize;                    // the size of the capture array
-        protected IDictionary Caps
-        {
```

Comment:
I know we just added these but for now I don't think we should remove them. Keep them for now but file an issue to follow-up on this as I would like to discuss it in FXDC.